File: cpp/easy/FindDisappearedNumbers.cpp
#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;
vector<int> findDisappearedNumbers(vector<int> &nums) {
  vector<int> map(nums.size(), 0);
  vector<int> rs;
  for (auto i : nums) {
    map[i - 1]++;
  }
  for (int i = 0; i < map.size(); ++i) {
    if (map[i] == 0) {
      rs.push_back(i + 1);
    }
  }
  return rs;
}

int main() {
  vector<int> nums = {1, 2, 3, 5, 6};
  findDisappearedNumbers(nums);
  cout << "fdsa" << endl;
  int aa;
  return 0;
}

File: rust/src/easy/find_disappeared_numbers.rs
pub fn find_disappeared_numbers(nums: Vec<i32>) -> Vec<i32> {
    let mut count:Vec<i32> = vec![0;nums.len()];
    for &val in &nums {
        count[(val-1) as usize] += 1;
    }
    let mut ans: Vec<i32> = Vec::new();
    for (index,&val) in count.iter().enumerate() {
        if val == 0 {
            ans.push((index + 1) as i32);
        }
    }
    ans
}